bpfman-operator/apis/v1alpha1/cluster_kprobe_program_types.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2023 The bpfman Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields are required unless explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha1
+
+// ClKprobeProgramInfo contains the information for the kprobe program
+type ClKprobeProgramInfo struct {
+	// The list of points to which the program should be attached.  The list items
+	// are optional and may be udated after the bpf program has been loaded
+	// +optional
+	// +kubebuilder:default:={}
+	Links []ClKprobeAttachInfo `json:"links"`
+}
+
+type ClKprobeAttachInfo struct {
+	// function to attach the kprobe to.
+	Function string `json:"function"`
+
+	// offset added to the address of the function for kprobe.
+	// The offset must be zero for kretprobes.
+	// TODO: Add a webhook to enforce kretprobe offset=0.
```

Comment:
Maybe this should be program type should use a distinct type? Maybe it wouldn't even make sense to have an `offset` field for this program type if it must be `0`?

Reply:
I think it serves the purpose better to have a separate type. Using a shared type ends up as more of a maintenance burden if the two implementations drift further apart in the future.

Reply:
done